### ISSUE TITLE
failed with exception: auth failure: unable to extract serviceURL

### ISSUE THREAD

**Issue by cabby26** [petergardfjall/garminexport#117]
Hello,

today i got this error.
failed with exception: auth failure: unable to extract serviceURL

**Comment by HeathVarmette** [comment]
Yes, I have run into the same issue. It occurs while attempting to pass login credentials using the browser impersonating installation
2024-09-22 11:21:36,011 [INFO] using 'curl_cffi' to create HTTP sessions that impersonate web browser 'chrome110' ...
2024-09-22 11:21:36,013 [INFO] authenticating user ...
2024-09-22 11:21:36,013 [INFO] passing login credentials ...
2024-09-22 11:21:36,358 [ERROR] failed with exception: auth failure: unable to extract serviceURL

**Comment by jesterpm** [comment]
Debug logs suggest that the client needs to provide a CAPTCHA.

```
2024-09-22 21:51:53,711 [DEBUG] got auth response 200: {"serviceURL":null,"serviceTicketId":null,"responseStatus":{"type":"CAPTCHA_REQUIRED","message":"","httpStatus":"OK"},"customerMfaInfo":null,"consentTypeList":null,"captchaAlreadyPassed":false}
2024-09-22 21:51:53,711 [ERROR] failed with exception: auth failure: unable to extract serviceURL
```

**Comment by Egregius** [comment]
Same problem here

**Comment by smoogle23** [comment]
same issue here as well, stop working on september 21st.

**Comment by drewbrew** [comment]
FWIW, the [garth approach](https://github.com/petergardfjall/garminexport/pull/114) is unaffected by the addition of the captcha

**Comment by Egregius** [comment]
Any updates on this? Quite a hasle to export all my activities manually.

**Comment by miguno** [comment]
+1, same problem

**Comment by Egregius** [comment]
I have become accustomed to exporting my activities manually. It would be nice to be able to do it automatically again, especially for holidays where we do several hikes a day.
I don't really understand the Garth approach nor what the problem is with the small json in a S3 bucket.

**Comment by ilsalgo** [comment]
+1 same here :(

**Comment by HrHorst** [comment]
Cloudflare captcha verification seems to be enforced. Sometimes I also have recaptcha as second verification.

For cloudflare capatcha this might be the only solution: https://github.com/FlareSolverr/FlareSolverr

**Comment by mymatenige** [comment]
Not the answer people are hoping for, but I've moved to https://github.com/pe-st/garmin-connect-export and it is working for me with near identical results.

**Comment by Egregius** [comment]
@mymatenige Thanks for the tip, up and running :)

**Comment by petergardfjall** [comment]
Sorry for being so unresponsive (I haven't had time nor energy to work on this for a while). I think, however, that I may have stumbled over an authentication flow that "just works" (without relying on browser impersonation etc). I would appreciate if anyone could try it out as per the instructions in the pull request https://github.com/petergardfjall/garminexport/pull/119.

**Comment by petergardfjall** [comment]
I've released a new version (`0.6.0`) which completely rewrites the authentication flow.
For me it works without requiring any use of bot protection or browser impersonation.
Please try it out and see if it works for you. Closing this ticket for now.

**Comment by miguno** [comment]
@petergardfjall : Thanks for the 0.6.0 release. It works again on my side!

**Comment by mymatenige** [comment]
Likewise. Thanks!
